shared-kernel-infrastructure/src/main/java/org/dcsa/skernel/infrastructure/pagination/Pagination.java

```diff
@@ -218,7 +218,7 @@ boolean isValidFilterParam(String key) {
 
       String[] keyParts = key.split(":");
       if (keyParts.length == 2) {
-        return filterParameters.contains(keyParts[0]) && ComparisonType.valueSet.contains(keyParts[1].toUpperCase());
+        return filterParameters.contains(keyParts[0]) && ComparisonType.VALUE_SET.contains(keyParts[1].toUpperCase());
```

Comment:
Open question: Should we allow the lower case variants (`asc`/`desc`) here? The Swagger only mentions the upper case variants.

Reply:
If not then we have to change the sorter as well, because Sorter.parseSort allows it, so if it passed the parseSort we can't filter it out here.

Anyway as we do support the uppercase variants I'd say we are compliant, I see no problem with allowing the lowercase variants as well.

Reply:
The "problem" is that we are building the reference implementation and we are now de facto deciding whether the lower case format is part of the standard or not.

Reply:
That is a separate issue in my opinion, I don't see this as the place to discuss that.

DCSAQueryParameterParser.parseComparisonType allows lowercase and so does Sorter.parseSort and the Pagination code cannot disallow what those two allows.

Maybe this is something to discuss in open issues? And if we do not wish to allow lowercase then it is a tech-debt issue in the other two classes.

Reply:
Pim reminded me that the URI is case-insensitive in general, so it should work with either.

Reply:
Order of things happening:

1. Parsing and validation of arguments
2. Querying the database
3. Constructing paging headers
4. Returning the result

The code you point to is part of step 3. Any discussion of weather we should allow lower case qualifiers should be around step 1. If step 1 allows lowercase (which it does) then step 3 absolutely also have to allow it.

Essentially the "de facto deciding" happened in may with this commit: https://github.com/dcsaorg/DCSA-Shared-Kernel/commit/56aa20de73293d9f4e912cf681720a952e57048c

So can we close the discussion here and move it somewhere more appropiate?

Reply:
It is already closed?